src/lua/LuaBody.cpp

```diff
@@ -630,28 +655,39 @@ static int l_body_distance_to(lua_State *l)
 static int l_body_get_ground_position(lua_State *l)
 {
 	Body *b = LuaObject<Body>::CheckFromLua(1);
+	AltitudeType altType = AltitudeType::DEFAULT;
+	bool terrainRelative;
+	if (!lua_isnoneornil(l, 2))
+	{
+		if (!pi_lua_strict_pull(l, 2, terrainRelative))
+		{
+			return luaL_error(l,
+				"invalid second argument passed. Should be \"true\" (above-terrain) or \"false\" (sea-level)"
+			);
+		}
+		altType = terrainRelative ? AltitudeType::ABOVE_TERRAIN : AltitudeType::SEA_LEVEL;
```

Comment:
Note: it's not required that the function strictly validate it's passed a boolean here. A simple `lua_toboolean` maintains the same API contract and is less expensive. This takes advantage of the language's "truthiness" feature and enables more flexible code without loosing correctness.

```suggestion
		altType = lua_toboolean(l, 2) ? AltitudeType::ABOVE_TERRAIN : AltitudeType::SEA_LEVEL;
```

Reply:
Changed to `lua_toboolean` instead of `pi_lua_strict_pull`. Performance is a concern here, espesially if this function is intended to be used in some UI later. And, really,  the only thing that prompted me to do first approach with `pi_lua_strict_pull` is to just notify user that he passed the wrong value.
~~I guess this will teach them the basics of implicit type conversion.~~

Reply:
Implicit truthiness is considered to be a very basic concept one learns in the very early stages of writing Lua - I don't think there's a serious concern here about unexpected behavior :smile: